shared/service_registry.go

```diff
@@ -23,6 +24,30 @@ type Service interface {
 	Status() error
 }
 
+// ServiceBase TODO
+type ServiceBase struct {
+	Ctx    context.Context    // Ctx TODO
```

Comment:
Ideally this should only be used directly by core functions like `Start`. Such core functions should pass the context as a parameter to all other functions, so that the ubiquitous `s.base.Ctx` invocation is replaced by `ctx`.